마이페이지 디자인 수정 (#122)

happydhKim opened the pull request:
# 마이페이지 디자인 수정 (#122)

## 해당 이슈 📎

- 마이페이지 디자인 수정 (#122)

## 변경 사항 🛠

- 마이페이지 신규 기획서에 맞게 디자인 수정.
- 채널 만료기간에 대한 더미 데이터로 대체.

## 테스트 ✨

> 없음

## 리뷰어 참고 사항 🙋‍♀️

-  채널 만료기간에 대한 더미 데이터로 대체. ( 추후 김김이조이 김재원 @load0ne 님의 feature로 대체 )

happydhKim committed:
[ FEATURE ] 마이페이지 스타일 리팩토링 : 유저 정보 (#122)

Why
- 마이페이지 스타일을 리팩토링 해야 한다.

What
- 마이페이지의 유저 정보의 연필 크기를 더 눈에 띄게 수정.
- 마이페이지의 width값을 조절, 폰트의 이모지 삭제.
- 히스토리 Card 크기 및 간격 재조정.

happydhKim committed:
[ FEATURE ] 마이페이지 히스토리 스타일 리팩토링 (#122)

Why
- 마이페이지의 히스토리 스타일을 리팩토링 해야 한다.
- 히스토리 카드에 채널 만료 날짜가 표기되어야 한다.
- 히스토리 카드에 채널 코드가 보여야 한다.

What
- 채널 만료 날짜와 채널 코드 구성 완료. ( 데이터는 아직 구현 안됨 )
- 히스토리 관련 리팩토링 완료.
- 마이페이지의 유저 정보 페이지 아바타 아래 간격 증가.

happydhKim committed:
[ REFACTOR ] 마이페이지는 로그인을 해야만 입장이 가능하다. (#122)

Why
- 익명의 사용자가 마이페이지에 들어갔을 때, 어색한 현상들이 많이 일어난다.

What
- 로그인을 하지 않은 사용자는 마이페이지에 들어오면 접근이 제한되도록 구현.
- 접근이 제한 된 것은 ErrorModal을 이용함.
- 에러메시지를 위해 constants 에 mypage 생성.

happydhKim committed:
[ FEATURE ] 마이페이지 채널코드 데이터 연동 (#122)

Why
- 마이페이지에 채널코드가 나와야 한다.

What
- 채널코드를 채널아이디를 이용해 구현.

happydhKim committed:
[ MERGE ] 최신버전으로 업데이트

Why
- 다른 작업 된 파일을 pull 받아야 한다.

What
- 최신버전으로 merge.

happydhKim committed:
[ REFACTOR ] 채널아이디 값이 없는 경우 처리 (#122)

Why
- 채널아이디 값이 없을 경우가 있을 수 있다.

What
- 채널 아이디가 null 인 경우를 처리.

happydhKim committed:
[ MERGE ] hotfix에 대한 충돌 수정 (#122)

Why
- 충돌을 해결해야 한다.

What
- 충돌 해결 후 최신 버전으로 업데이트.

devjaewon commented on web/src/constants/mypage.js:
```diff
@@ -0,0 +1,3 @@
+const LOGIN_STATUS_ERROR_MESSAGE = '로그인을 해주셔야 이용이 가능합니다.';
```
constants 는 앞으로 여러개가 생길 가능성이 높은 파일로 보이는데, `export` 문으로 하는게 좋지 않을까요?
eslint 에 걸린다면 일단 주석으로 disable 해두는 것이 좋을 것 같습니다.

happydhKim replied:
오! 맞아요 eslint 때문에 하나만 일단 작업을 해뒀습니다. 다른 문구가 있는지 확인해보고 정리를 할게요 !!🤠

devjaewon reviewed (approved):
금방 작업하셨네요! 고생하셨습니다 👍

always-awake commented on web/src/components/history/UserHistory/UserHistory.jsx:
```diff
@@ -35,7 +35,7 @@ const UserHistory = (props) => {
       <S.Alert>
         <span aria-label="really" role="img">🤭</span>
-        아직 채널을 한번도 생성 안해보셨네요??
+        기록이 없어요.
```
이 텍스트도 span 태그로 감싸면 좋을거 같아요~

happydhKim replied:
넹 알겠습니당 !!🐤

always-awake reviewed (commented):
수고 많으셨습니당 🙆‍♀️

aereeeee commented on web/src/components/history/UserHistoryCard/UserHistoryCard.jsx:
```diff
@@ -13,20 +13,21 @@ const UserHistoryCard = (props) => {
     channelName,
     displayName,
   } = props;
+  const channelCode = channelId !== null ? channelId.substring(0, 5) : 'null';
```
채널 코드에 문자열로 null 들어가면 안될것같은데요!

happydhKim committed:
Merge branch 'develop' of https://github.com/connect-foundation/2019-02 into feature/issue122

happydhKim committed:
[ REFACTOR ] 리뷰 받은 사항 적용 (#122)

Why
- 리뷰 받은 내용을 수정해야 한다.

What
- 일반 폰트에 span태그 추가.
- constants에 컨벤션 적용.
- 리스너 히스토리 0인 경우 안내 문구가 안나오던 현상 수정.
